ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/TerminateQueryExecutor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.parser.tree.TerminateQuery;
+import io.confluent.ksql.rest.SessionProperties;
+import io.confluent.ksql.rest.entity.DropConnectorEntity;
+import io.confluent.ksql.rest.entity.ErrorEntity;
+import io.confluent.ksql.rest.entity.KsqlEntity;
+import io.confluent.ksql.rest.entity.WarningEntity;
+import io.confluent.ksql.rest.server.computation.DistributingExecutor;
+import io.confluent.ksql.rest.server.computation.ValidatedCommandFactory;
+import io.confluent.ksql.services.ConnectClient.ConnectResponse;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import java.util.Optional;
+import org.apache.hc.core5.http.HttpStatus;
+
+public final class TerminateQueryExecutor {
+
+  private TerminateQueryExecutor() { }
+
+  public static Optional<KsqlEntity> execute(
+      final ConfiguredStatement<TerminateQuery> statement,
+      final SessionProperties sessionProperties,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext
+  ) {
+
+    if (statement.getStatementText().contains("transient")){
+      executionContext.getQuery(statement.getStatement().getQueryId().get()).get().close();
+      return Optional.of(new TerminateQueryEntity(statement.getStatementText(), connectorName));
+    } else {
+      // do default behaviour for terminating persistent queries
```

Comment:
@AlanConfluent I am having trouble trying to implement this part cleanly. Ideally I want to just break out of here and default to the default executor but I'm not sure if I can do that. Otherwise, I could just reimplement all of the default logic with the distributor here, but it seems messy to me, what do you think?

Reply:
We chatted a bit about this offline. It would have been nice if rather than mapping a class to static method references, these were instead instances such that you could inject in the `DistributingExecutor` in the constructor and could even have a base parent class that could invoke the default behavior so that it could be shared between all of the custom executors.  But that's a larger refactoring than we require now.

As we discussed before I think the most straightforward solution is to add `DistributingExecutor` as a parameter of execute and write it to command topic explicitly here.  That means you'd have to add it to the interface `StatementExecutor` and then all of the conforming classes as well.

Reply:
> add DistributingExecutor as a parameter of execute and write it to command topic explicitly here. That means you'd have to add it to the interface StatementExecutor and then all of the conforming classes as well.

I feel like that breaks the abstraction boundaries set up here. What do you think of (and this requires the first change you suggested) having executors have an optional `isApplicable` method or something like that which can cause the system to default to the distributing executor if the condition doesn't match?

Reply:
That was another change that we talked about doing.  We could either do that, or even have the return type be another object wrapping the return value and having a flag similar to the one you proposed (e.g. `StatementExecutorResponse.handled(Optional.of(entity))` or `StatementExecutorResponse.notHandled()`).

I can appreciate if you think these custom executors shouldn't know about a fallback strategy and I agree it keeps them simpler.

Sound reasonable @nateab ? Sorry if you're going to have to make a lot of small changes to the handlers again.

Reply:
Sounds good to me! And no problem haha, having cleaner code is worth it